app/src/main/java/com/example/weather/ui/forecast/ForecastItemAdapter.java

```diff
@@ -51,11 +66,22 @@ public MyViewHolder(ItemForecastBinding itemForecastBinding) {
         }
 
 
+        @SuppressLint("CheckResult")
+        @RequiresApi(api = Build.VERSION_CODES.N)
         public void bind(WeatherData weatherData, int position) {
             this.binding.tvIndex.setText((position+1)+"");
             this.binding.tvCity.setText(weatherData.getName());
             this.binding.layout.setOnClickListener(new onClickListener(weatherData.getId()));
-            viewModel.getCurrentWeatherByCoord(weatherData.getLan(),weatherData.getLon(),this.binding.tvCurrentTemp);
+            viewModel.getCurrentWeatherByCoord(weatherData.getLan(),weatherData.getLon())
```

Comment:
ViewHolder shouldn't make request. Fragment should invoke this method. Request via RxJava should be inside ViewModel, ViewModel should update LiveData, LiveData should update Fragment, Fragment should update RecyclerView